consensus/ibft/consensus_backend.go

```diff
@@ -64,6 +70,21 @@ func (i *backendIBFT) InsertBlock(
 		return
 	}
 
+	// WriteCommittedSeals alters the extra data before writing the block
+	// It doesn't handle errors while pushing changes which can result in
+	// corrupted extra data.
+	// We don't know exact circumstance of the unmarshalRLP error
+	// This is a safety net to help us narrow down and also recover before
+	// writing the block
+	if err := i.ValidateExtraDataFormat(newBlock.Header); err != nil {
+		i.logger.Error("cannot write block: corrupted extra data",
```

Comment:
could we use fmt.Sprintf?

Reply:
What I can do is rearrange the args for err to maybe be on the last position and more conform to what @ferranbt linked. If the reasoning for Sprintf is because of the conversion of byte array to hex, as far as i searched, its considerably slower than doing it by hand.

If this is only motivation can we maybe keep it as is?

@ferranbt please let me know if your comment is a remark to my code or to @JekaMas comment

Reply:
I am more about "\n\t" - that kind of things. Could we either get rid of them or think about formatting?

Reply:
Yes we can. Will be removed.